include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1673,117 +1549,91 @@ struct __parallel_sort_submitter<__internal::__optional_kernel_name<_LeafSortNam
                                  __internal::__optional_kernel_name<_GlobalSortName...>,
                                  __internal::__optional_kernel_name<_CopyBackName...>>
 {
-    template <typename _ExecutionPolicy, typename _Range, typename _Merge, typename _Compare>
+    template <typename _ExecutionPolicy, typename _Range, typename _Compare>
     auto
-    operator()(_ExecutionPolicy&& __exec, _Range&& __rng, _Merge __merge, _Compare __comp) const
+    operator()(_ExecutionPolicy&& __exec, _Range&& __rng, _Compare __comp) const
     {
         using _Policy = typename ::std::decay<_ExecutionPolicy>::type;
         using _Tp = oneapi::dpl::__internal::__value_t<_Range>;
         using _Size = oneapi::dpl::__internal::__difference_t<_Range>;
 
-        _Size __n = __rng.size();
+        const ::std::size_t __n = __rng.size();
         assert(__n > 1);
 
-        _PRINT_INFO_IN_DEBUG_MODE(__exec);
-
-        // __leaf: size of a block to sort using algorithm suitable for small sequences
-        // __optimal_chunk: best size of a block to merge duiring a step of the merge sort algorithm
-        // The coefficients were found experimentally
-        _Size __leaf = 4;
-        _Size __optimal_chunk = 4;
-        if (__exec.queue().get_device().is_cpu())
-        {
-            __leaf = 16;
-            __optimal_chunk = 32;
-        }
-        // Assume powers of 2
-        assert((__leaf & (__leaf - 1)) == 0);
-        assert((__optimal_chunk & (__optimal_chunk - 1)) == 0);
-
-        const _Size __leaf_steps = ((__n - 1) / __leaf) + 1;
+        const ::std::uint16_t __chunk = 4;
+        const _Size __steps = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __chunk);
 
         // 1. Perform sorting of the leaves of the merge sort tree
-        sycl::event __event1 = __exec.queue().submit([&](sycl::handler& __cgh) {
+        sycl::event __event1 = __exec.queue().submit([&, __comp](sycl::handler& __cgh) {
```

Comment:
Everywhere in std algorithm a predicate in passed by value. So, one reason  - to make the code consistent.
BTW, theoretically for parallel computation, compiler has more possibilities to generate effective code when each thread has  separate copy of variable, (not shared variable); plus no sharing variables eliminates side effect (races ) in case of predicate stateful  (of course a caller is responsible for correctness in any way). It is the second reason.